src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const item = this[i];
+
+      if (typeof item === 'string'
+        || typeof item === 'number'
+        || typeof item === 'boolean') {
+        str += '' + separator + item;
+      } else {
+        str += '' + separator;
```

Comment:
you can check
```
if (i < this.length - 1) {
  str += separator
}
```

Reply:
it s have no sence

Reply:
why do you think so? if you have questions, ask in the chat